[FIX] Minor text margin adjustments for toast

Polysemouse opened the pull request:
# Description

- move toast to bottom left of screen so it won't block modals as much for smaller screens

Before|Afte (Mobile)|After (PC)
---|---|---
![image](https://user-images.githubusercontent.com/107602352/211737455-f511d4d8-5df3-4b1b-8455-339ef0dec4e1.png)|![image](https://user-images.githubusercontent.com/107602352/216218037-7daa967b-3912-44f2-8975-1c74e21b24f3.png)|![image](https://user-images.githubusercontent.com/107602352/216218054-b67fddad-a424-4ab8-ab2b-7dbb70212e55.png)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

spencerdezartsmith commented:
Played around with this a little but don't think this position is the best one. It solves the issue related to the stock label but it causes overlaps on mobile see screenshot. 

Some other possible solutions:

- Bottom left. The actions could grow up vertically. So the first one would be bottom 0 or whatever and then the next would push the last one up. Not sure if that makes sense. 
- Make the toast smaller on mobile and reduce the number of actions shown at one time?? Also, combine the same actions rather than adding a duplicate toast. 

Not sure what is the best solution right now but I don't think this one is the one.

<img width="436" alt="Screen Shot 2023-01-14 at 10 05 16 pm" src="https://user-images.githubusercontent.com/17863697/212468913-f68e8915-6afc-45c6-bc36-cace1965fdfe.png">

Polysemouse commented:
@spencerdezartsmith I moved the toast to the bottom left of the screen above the farm ID based on your suggestion.  I also reduced the font size for mobile.

>Make the toast smaller on mobile and reduce the number of actions shown at one time?? Also, combine the same actions rather than adding a duplicate toast.
This would be the best approach, however it requires using a completely different implementation.  Currently the toasts are set with an image and string across different files.  This new approach requires a listener listening to changes to the player's inventory all the time and setting the toast according to the changes.  I haven't been able to figure that part yet, so I am using your first suggestion for now.

AzaroQL commented:
I think its better to preserve old location(top left), but combine toasts. It will save space and also provide better UX to user since he will see how much he spent on seeds(in that particular case). 
Displaying 5 identical messages provide no use for player.

Polysemouse commented:
@AzaroQL Combining toasts would require reimplementing how toasts work behind the scenes (perhaps the changes of the whole inventory state need to be monitored).  For now would it be better to keep the toast in the original position for PC and the bottom left position for mobile as a temporary measure?

c-py commented:
> @AzaroQL Combining toasts would require reimplementing how toasts work behind the scenes (perhaps the changes of the whole inventory state need to be monitored). For now would it be better to keep the toast in the original position for PC and the bottom left position for mobile as a temporary measure?

@Polysemouse I like it down the bottom left for the mobile. It looks nice. 

> For now would it be better to keep the toast in the original position for PC and the bottom left position for mobile as a temporary measure?

I agree for this, on PC top left looks nice. Would it be possible to change the position to bottom left on mobile and top left on PC?

Also I think we should keep the text size the same as the Land ID label. Would it be possible to revert this part.

c-py commented:
I also love the `mb-0.5` you put on the Balance label in the feed bumpkin modal - I think the toast could use this as well 😁

Polysemouse commented:
@c-py Thanks for the suggestions!  I made the following changes to the PR:
- Revert the text size back to `text-sm` for all screen sizes
- add the `mb-0.5` to text
- keep original position and toast sorting order for large screens; put toasts at the bottom of the screen and reverse toast sorting order for small screens

spencerdezartsmith commented:
> @AzaroQL Combining toasts would require reimplementing how toasts work behind the scenes (perhaps the changes of the whole inventory state need to be monitored). For now would it be better to keep the toast in the original position for PC and the bottom left position for mobile as a temporary measure?

I agree with this. I personally don't love the new position on mobile. I think I would prefer to see the blocked label moved to the right before implementing an entirely new position for the toast.

Polysemouse commented:
@spencerdezartsmith I moved the label to the center for the custom component in https://github.com/sunflower-land/sunflower-land/pull/2159.  Going forward other crafting modals will use that component so hope that will solve the overlapping toast issue instead.

I reverted the changes to the mobile position so this PR would just be some very minor text position adjustments

spencerdezartsmith reviewed (approved):
LGTM
